main.cpp

```diff
@@ -427,12 +551,94 @@ bool Bicycle::brake(int appliedPressure)
 {
     return appliedPressure > 0 ? true : false;
 }
+void Bicycle::ringBell()
+{
+    std::cout << "Ding, Ding \n";
+}
 
-
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 int main()
 {
-    Example::main();
+    // Example::main();
+    
+    Computer computer;
+    computer.connectToInternet();
+    computer.save();
+    computer.runApp();
+    std::cout << "Is the RAM higher than 8? " << (computer.RAMAmount > 8 ? "Yes" : "No") << "\n\n";
+    
+    Car car;
+    car.moveForward();
+    car.stopMovement();
+    car.turn();
+    car.honk();
+    std::cout << "Does it have 5 seats? " << (car.numberOfSeats >= 5 ? "Yes" : "No") << "\n\n";
+    
+    Tree tree;
+    tree.grow();
+    // This produceFruit is giving me issues again.  I do not understand the error it produces when I run it:
+        // in function `main':
+        // main.cpp:(.text+0x16e2): undefined reference to `Tree::produceFruit()'
```

Comment:
You **declared** a function.
You **called** that function somewhere.
You **did not define** that function. 
That is what **undefined** is referring to in this error message.

You need to start establishing the habit of searching for these types of error messages so you can figure out the solution yourself. 
https://www.google.com/search?q=cplusplus+undefined+reference+to&oq=cplus+plus+undeined&aqs=chrome.1.69i57j0i13i30j0i390l4.3734j0j7&sourceid=chrome&ie=UTF-8
- [ ] revise

Reply:
My mistake was on line 199.  I was confused because I was defining here,  I just wasn't naming it properly